### ISSUE TITLE
CUDA error on ubuntu system

### ISSUE THREAD

**Issue by nirajvermafcb** [AlexeyAB/darknet#378]
Hi, I am using ubuntu 16.04 with nvidia GEFORCE 920 gpu and I have CUDA version 7.5 installed on my computer.
I have made the necessary changes in make file replacing the values of **GPU** and **CUDNN**
as 1.And then when I run **make** command it gives me this error.

> fatal error: cudnn.h: No such file or directory
compilation terminated.
Makefile:89: recipe for target 'obj/gemm.o' failed
make: *** [obj/gemm.o] Error 1

 When i don't make this changes it just works fine..

**Comment by TheMikeyR** [comment]
Did you install cudnn correctly? It seems that it can't find cudnn.h file, which you are getting when installing cudnn. Try to compile with Cudnn=0 and see if that works.
You can check this answer on [how to install cudnn](https://askubuntu.com/a/767270/228546) and do these steps and try to remake.

**Comment by nirajvermafcb** [comment]
yes with Cudnn = 0 it is working but I want to run it with GPU.

**Comment by TheMikeyR** [comment]
GPU is 1, so you are running it with GPU. 
cuDNN is just a library which adds faster computation for deep learning operations 
> is a GPU-accelerated library of primitives for deep neural networks. cuDNN provides highly tuned implementations for standard routines such as forward and backward convolution, pooling, normalization, and activation layers. cuDNN is part of the NVIDIA Deep Learning SDK. [SOURCE](https://developer.nvidia.com/cudnn)

If you want to run with cuDNN on you should install like it's described in the link in my previous comment or you can read below, copied from [source](https://askubuntu.com/a/767270/228546), You can skip step 0, since you already installed CUDA.:  

**Step 0:** Install cuda from the standard repositories.  (See [How can I install CUDA on Ubuntu 16.04?](https://askubuntu.com/q/799184/10425))

**Step 1:** Register an nvidia developer account and [download cudnn here](https://developer.nvidia.com/cudnn) (about 80 MB)

**Step 2:** Check where your cuda installation is. For the installation from the repository it is `/usr/lib/...` and `/usr/include`. Otherwise, it will be `/usr/local/cuda/` or `/usr/local/cuda-<version>`. You can check it with `which nvcc` or `ldconfig -p | grep cuda`

**Step 3:** Copy the files:

    $ cd folder/extracted/contents
    $ sudo cp -P include/cudnn.h /usr/include
    $ sudo cp -P lib64/libcudnn* /usr/lib/x86_64-linux-gnu/
    $ sudo chmod a+r /usr/lib/x86_64-linux-gnu/libcudnn*

Then recompile your darknet with CUDNN=1, and it should complete the compilation if you followed step 0-3, I assume you have installed CUDA drivers already since you can compile with GPU=1 so you can just go to step 1-3.

**Comment by nirajvermafcb** [comment]
I uninstalled 7.5 version and again installed 9.1 version from nvidia website with all updated drivers as per my system.It is still giving me the same error.

**Comment by AlexeyAB** [comment]
@nirajvermafcb Did you use in the Makefile?
```
GPU=1
CUDNN=0
```

**Comment by nirajvermafcb** [comment]
Yes with GPU = 1 and CUDNN = 0 it is running fine but when I set CUDNN = 1 it is showing me this error @AlexeyAB

**Comment by AlexeyAB** [comment]
@nirajvermafcb Did you do all **3 steps (1, 2, 3)** that written by TheMikeyR: https://github.com/AlexeyAB/darknet/issues/378#issuecomment-363701504

**Comment by SJRogue** [comment]
I've had this problem as well until i finally did the installation of cuDNN correctly, also the referring.

Be carefull you're saying u have version 9.1 cuDNN, do you have the right CUDA/cuDNN match?
